lib/collection/src/collection/distance_matrix.rs

```diff
@@ -12,26 +16,140 @@ use crate::operations::types::{CollectionResult, CoreSearchRequest, CoreSearchRe
 use crate::operations::universal_query::shard_query::{Sample, ScoringQuery, ShardQueryRequest};
 
 #[derive(Debug, Default)]
-pub struct DistanceMatrixResponse {
-    pub sample_ids: Vec<PointIdType>,   // sampled point ids
-    pub nearest: Vec<Vec<ScoredPoint>>, // nearest points for each sampled point
+pub struct CollectionSearchMatrixResponse {
+    pub sample_ids: Vec<PointIdType>,    // sampled point ids
+    pub nearests: Vec<Vec<ScoredPoint>>, // nearest points for each sampled point
```

Comment:
If order of ids is not important as an internal representation, this can be represented as a HashMap. It would make it easier to understand, IMO